src/Compilers/CSharp/Portable/Symbols/BaseTypeAnalysis.cs

```diff
@@ -190,10 +190,11 @@ private static (bool definitelyManaged, bool hasGenerics) DependsOnDefinitelyMan
                         continue;
                     }
 
+                    fieldType.AddUseSiteDiagnostics(ref useSiteDiagnostics);
```

Comment:
Does this mean that we will get say obsolete warnings when we use a `struct` in a place where we checked the `unmanaged` constraint?

Reply:
> Does this mean that we will get say obsolete warnings when we use a struct in a place where we checked the unmanaged constraint? 

Obsolete warnings are not reported through the use-site diagnostics mechanism. 

---
In reply to: [460266173](https://github.com/dotnet/roslyn/pull/46114#discussion_r460266173) [](ancestors = 460266173)

Reply:
Reference unification warnings could be reported

---
In reply to: [460271139](https://github.com/dotnet/roslyn/pull/46114#discussion_r460271139) [](ancestors = 460271139,460266173)